### PR TITLE
Add to have focus

### LINKED ISSUE #61: (feature request): Adding .toHaveFocus() to the library
Hello folks and thank you for the great work you're doing with the tool! That's amazing 😊 

I'm opening this issue to request for a `.toHaveFocus()` function that will allow to verify what is the actually focused element. This may become very helpful when testing for accessibility.

Actually, to determine if an element has focused on it, I have two options that are quite verbose:

```js
  const axTreeSnapshot = await page.accessibility.snapshot({ root });
  // does not look very consistent on firefox from my experimentation, but I may be wrong
  axTreeSnapshot.focused
```

or as described in https://github.com/microsoft/playwright/issues/2159#issuecomment-625883641 with the following:

```js
const foo1Isfocused = await page.$eval("#foo1", (el) => el === document.activeElement)
```

What do you think?

### PR DESCRIPTION
**Pull request by mmarkelov**
close #61

### CODE REVIEW

File: global.d.ts [hunk 1]
```diff
@@ -39,10 +39,13 @@ export interface PlaywrightMatchers<R> {
   */
   toEqualText(value: string, options?: PageWaitForSelectorOptions): Promise<R>;
   /**
-  * Will ensure that the element is one the page in a given timeout (default 1 second).
+  * Will ensure that the element is on the page in a given timeout (default 1 second).
```
Comment (mxschmitt) [inline global.d.ts:42]:
```suggestion
  * Will ensure that the element is on the page.
```
we removed the default 1 second timeout a while ago if I'm correct. There are other places in the readme 

> By default it waits 1 second for the element which you can overwrite via the options.

3 times, would be great if you could remove them too.

File: global.d.ts [hunk 2]
```diff
@@ -39,10 +39,13 @@ export interface PlaywrightMatchers<R> {
...
   */
   toHaveSelector(selector: string, options?: PageWaitForSelectorOptions): Promise<R>;
-
+  /**
+   * Will ensure that the element is in focus.
```
Comment (mxschmitt) [inline global.d.ts:46]:
```suggestion
   * Will ensure that the element has focus.
```

File: src/matchers/toHaveFocus/index.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    const isFocused = await page.$eval(selector, (el) => el === document.activeElement)
+    if (isFocused) {
+      return {
+        pass: true,
+        message: () => `${quote(selector)} has focus on it.`
+      }
+    }
+    return {
+      pass: false,
+      message: () => `${quote(selector)} has not focus on it.`
```
Comment (mxschmitt) [inline src/matchers/toHaveFocus/index.ts:19]:
```suggestion
      message: () => `${quote(selector)} has no focus on it.`
```

**Review by mxschmitt — APPROVED** [review]

File: global.d.ts (hunk 1 above)
Comment (mmarkelov) [reply global.d.ts]: done

**Pull request merged**